ex04.ex

```diff
@@ -75,10 +66,14 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
-
-
-
+  def even_odd(x) do
```

Comment:
### -1

You clearly get the code, so I'm going to focus on the more detailed stuff here. In particular, we have a six line function, which is pushing it. It has a local variable, which is also a warning sign.

Never be afraid to split into functions:

~~~ elixir
  def even_odd(list) do
    reduce(list, { [], [] }, &partition_even_odd/2)
    |> reverse_two_lists
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}

  defp reverse_two_lists({ evens, odds }) do
    { reverse(evens), reverse(odds) }
  end
~~~